script.js

```diff
@@ -38,7 +38,7 @@ export async function script(octokit, repository, options) {
         };
       },
       (error) => {
-        if (error.status === 404) return { pkg: false };
+        if (error.status === 404) return { pkg: false, sha: undefined };
```

Comment:
Actually can you add a comment to explain why we did this? Maybe it will become obsolete in future. Maybe sth like this

```js
         // return { sha } to make TypeScript check happy
         if (error.status === 404) return { pkg: false, sha: undefined };
```

Reply:
I was thinking... Should this be an issue to TypeScript and we should use `//@ts-expect-error` here? @gr2m (adding the comment you suggested in the meantime)

Reply:
yes I would appreciate if you could check in with the TypeScript folks whether this is a bug or expected behavior. 

Actually using `// @ts-expect-error` is a good idea, it's more elegant, and I think TypeScript will complain once there no longer is an error. So maybe do this?

```js
         // @ts-expect-error expects { sha} to be present (see #6)
         if (error.status === 404) return { pkg: false };
```

Reply:
https://github.com/octoherd/script-setup-renovate/pull/10